### DISCUSSION TITLE
How can I tell if a page is loaded directly or is navigated to within the app?

### DISCUSSION THREAD

**Discussion by mildfuzz** [vercel/next.js#22289] [Help]
So my issue is that I can not allow the user to refresh without re-fetching certain bits of data. We do this within getServerSideProps on each page

Is there a way I can tell if a the user is loading the page fresh (either through direct navigation or refreshing the page) or has navigated within the app and will therefore have local state (and thus not need to re-run the xhr)

**Comment by michrome** [comment]
I'm not sure this is helpful, but I'm wondering: might it be more reliable to check the local state rather than how the page was loaded? If the state is missing then you could use something like [SWR](https://swr.vercel.app) to populate it.
